components/kyma-environment-broker/internal/ias/client.go

```diff
@@ -0,0 +1,231 @@
+package ias
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	kebError "github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/error"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	PathServiceProviders  = "/service/sps"
+	PathCompanyGlobal     = "/service/company/global"
+	PathAccess            = "/service/sps/%s/rba"
+	PathIdentityProviders = "/service/idp"
+	PathDelete            = "/service/sps/delete"
+)
+
+type (
+	ClientConfig struct {
+		URL    string
+		ID     string
+		Secret string
+	}
+
+	Client struct {
+		config         ClientConfig
+		httpClient     *http.Client
+		closeBodyError error
+	}
+
+	Request struct {
+		Method  string
+		Path    string
+		Body    io.Reader
+		Headers map[string]string
+	}
+)
+
+func NewClient(cli *http.Client, cfg ClientConfig) *Client {
+	return &Client{
+		config:     cfg,
+		httpClient: cli,
+	}
+}
+
+func (c Client) CloseBodyError() error {
```

Comment:
it is not used anywhere. IMO we should get rid of it as it's quite not convenient.  The better option is to just use the append error, see my example in `do` method

Reply:
removed